cmd/clusterctl/client/cluster/topology.go

```diff
@@ -251,15 +265,21 @@ func (t *topologyClient) prepareInput(ctx context.Context, in *DryRunInput, apiR
 
 // setMissingNamespaces sets the object to the current namespace on objects
 // that are missing the namespace field.
-func (t *topologyClient) setMissingNamespaces(in *DryRunInput) error {
-	currentNamespace := metav1.NamespaceDefault
-	// If a cluster is available use the current namespace as defined in its kubeconfig.
-	if err := t.proxy.CheckClusterAvailable(); err == nil {
-		currentNamespace, err = t.proxy.CurrentNamespace()
-		if err != nil {
-			return errors.Wrap(err, "failed to get current namespace")
+func (t *topologyClient) setMissingNamespaces(in *TopologyPlanInput) error {
+	// If TargetNamespace is provided use it.
+	currentNamespace := in.TargetNamespace
+	if currentNamespace == "" {
+		// If TargetNamespace is not provided use "default" namespace.
+		currentNamespace = metav1.NamespaceDefault
+		// If a cluster is available use the current namespace as defined in its kubeconfig.
+		if err := t.proxy.CheckClusterAvailable(); err == nil {
+			currentNamespace, err = t.proxy.CurrentNamespace()
+			if err != nil {
+				return errors.Wrap(err, "failed to get current namespace")
+			}
```

Comment:
Let's move this in default and validate

Reply:
`defaultAndValidate` is used to call the webhooks on the `Cluster` and `ClusterClass` types. However, this piece of code is for preprocessing all the inputs (analogous to how kubectl would set the namespaces). IMHO, this logic should sit under the `prepareInput` flow.

Reply:
Given the current structure with prepareInput, I would also prefer keeping it here. 

Should we maybe rename `defaultAndValidate` to e.g. `runDefaultAndValidationWebhooks` to make easy to see at the first glance what it does and what its scope is?